packages/neuron-wallet/src/services/indexer/queue.ts

```diff
@@ -0,0 +1,221 @@
+import { Subject, Subscription } from 'rxjs'
+import Utils from 'services/sync/utils'
+import logger from 'utils/logger'
+import GetBlocks from 'services/sync/get-blocks'
+import { Transaction } from 'types/cell-types'
+import TypeConvert from 'types/type-convert'
+import BlockNumber from 'services/sync/block-number'
+import AddressesUsedSubject from 'models/subjects/addresses-used-subject'
+import LockUtils from 'models/lock-utils'
+import TransactionPersistor from 'services/tx/transaction-persistor'
+import IndexerTransaction from 'services/tx/indexer-transaction'
+
+import IndexerRPC from './indexer-rpc'
+
+enum TxPointType {
+  CreatedBy = 'createdBy',
+  ConsumedBy = 'consumedBy',
+}
+
+export default class Queue {
```

Comment:
While I understand we have different folders - to some extend they act like namespaces, having multiple `Queue` classes worries me that it could confuse reading and understanding our design and implementation.

Reply:
How about `IndexerQueue`

Reply:
That's better. We have two queue implementations for two different modules, then we name them differently for clarification. Please do so.